controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java

```diff
@@ -141,7 +142,40 @@ private void authenticate(String resourceName, AuthHandler.Permissions level) th
             String credentials = authParams.get(0);
             assert credentials != null;
 
-            if (!pravegaAuthManager.authenticate(resourceName, credentials, level)) {
+            if (!pravegaAuthManager.authenticateAuthorize(resourceName, credentials, level)) {
+                throw new AuthorizationException("Auth failed for " + resourceName, 403);
+            }
+        }
+    }
+
+    private Principal authenticate() throws AuthException {
+        if (pravegaAuthManager != null ) {
+            List<String> authParams = headers.getRequestHeader(HttpHeaders.AUTHORIZATION);
+
+            if (authParams == null || authParams.isEmpty()) {
+                throw new AuthenticationException("Auth failed.");
+            }
+
+            String credentials = authParams.get(0);
+            assert credentials != null;
+
+            return pravegaAuthManager.authenticate(credentials);
+        }
+        return null;
+    }
+
+    private void authorize(String resourceName, Principal principal, AuthHandler.Permissions level) throws AuthException {
+        if (pravegaAuthManager != null ) {
+            List<String> authParams = headers.getRequestHeader(HttpHeaders.AUTHORIZATION);
+
+            if (authParams == null || authParams.isEmpty()) {
+                throw new AuthenticationException("Auth failed for " + resourceName);
+            }
+
+            String credentials = authParams.get(0);
+            assert credentials != null;
```

Comment:
Why is this an assert? Isn't this a parameter, in which case it should be a precondition check?

Reply:
This is not a parameter. Replaced assert with a `Preconditions.notNull` check.

Reply:
I believe this is still there.

Why would it not be considered a parameter if you fetch it from `authParams` which is retrieved externally?

Reply:
Fixed. Using `PreConditions`.